packages/applicant/src/pages/submissions/[submissionId]/summary.page.tsx

```diff
@@ -0,0 +1,275 @@
+import { GetServerSideProps } from 'next';
+import getConfig from 'next/config';
+import Link from 'next/link';
+import Layout from '../../../components/partials/Layout';
+import Meta from '../../../components/partials/Meta';
+import {
+  getQuestionById,
+  getSubmissionById,
+  hasSubmissionBeenSubmitted,
+  QuestionType,
+  SectionData,
+} from '../../../services/SubmissionService';
+import { getApplicationStatusBySchemeId } from '../../../services/ApplicationService';
+import { GrantMandatoryQuestionService } from '../../../services/GrantMandatoryQuestionService';
+import { initiateCSRFCookie } from '../../../utils/csrf';
+import { getJwtFromCookies } from '../../../utils/jwt';
+import { routes } from '../../../utils/routes';
+import { ProcessMultiResponse } from './sections/[sectionId]/processMultiResponse';
+
+const { publicRuntimeConfig } = getConfig();
+
+export interface SubmissionSummaryPage {
+  sections: SectionData[];
+  grantSubmissionId: string;
+  mandatoryQuestionId: string;
+  applicationName: string;
+  hasSubmissionBeenSubmitted: boolean;
+  csrfToken: string;
+}
+
+export const getServerSideProps: GetServerSideProps<
+  SubmissionSummaryPage
+> = async ({ req, res, params }) => {
+  const jwt = getJwtFromCookies(req);
+  const submissionId = params.submissionId.toString();
+
+  const { sections, grantSubmissionId, applicationName, grantSchemeId } =
+    await getSubmissionById(submissionId, getJwtFromCookies(req));
+
+  const grantApplicationStatus = await getApplicationStatusBySchemeId(
+    grantSchemeId,
+    jwt
+  );
+
+  if (grantApplicationStatus === 'REMOVED') {
+    return {
+      redirect: {
+        destination: `/grant-is-closed`,
+        permanent: false,
+      },
+    };
+  }
+
+  const hasBeenSubmitted = await hasSubmissionBeenSubmitted(submissionId, jwt);
+
+  await initiateCSRFCookie(req, res);
+
+  const hydratedSections = await Promise.all(
+    sections.map(async (section) => {
+      return {
+        ...section,
+        questions: await Promise.all(
+          section.questionIds.map(async (questionId) => {
+            const questionData = await getQuestionById(
+              submissionId,
+              section.sectionId,
+              questionId,
+              jwt
+            );
+            return questionData.question;
+          })
+        ),
+      };
+    })
+  );
+
+  const mandatoryQuestionService = GrantMandatoryQuestionService.getInstance();
+  const mandatoryQuestionDto =
+    await mandatoryQuestionService.getMandatoryQuestionBySubmissionId(
+      submissionId,
+      jwt
+    );
+  const mandatoryQuestionId = mandatoryQuestionDto.id;
```

Comment:
Unless I'm missing something we pass this `mandatoryQuestionId` down to `QuestionRow` but we never actually do anything with it?

Reply:
we will be as part of GAP-2096: https://github.com/cabinetoffice/gap-find-apply-web/pull/234/files#diff-541358a63993057c50641472fe8160e1e0ab9f51b9c1427200657f189c4c20e0R275

I was working on them in tandem so there's a bit of overlap between the two. I'm happy to pull this out here if you want though

Reply:
Ahh okay no worries if it's gonna be used for something.